### PR TITLE
Feat/cloudflare/audit logs

### PR DESCRIPTION
**Pull request by Adamowoc**
Audit logs part of the Cloudflare zerotrust parser.

Related to : https://github.com/SekoiaLab/platform/issues/38299

### CODE REVIEW

File: Cloudflare/audit_logs/ingest/parser.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  - name: set_cloudflare_fields
+
+stages:
+  set_ecs_fields:
+    actions:
+      - set:
+          "@timestamp": '{{json_event.message.When}}'
+          user.email: '{{json_event.message.ActorEmail}}'
+          user.id: '{{json_event.message.ActorID}}'
+          source.ip: '{{json_event.message.ActorIP}}'
```
Comment (squioc) [inline Cloudflare/audit_logs/ingest/parser.yml:18]: I suggest adding the following fields
```suggestion
          source.ip: '{{json_event.message.ActorIP}}'
          event.category: ["network"]
          event.dataset: "audit_logs"
          event.kind: 'event'
          event.type: ["access"]
          observer.type: "proxy"
          observer.vendor: "Cloudflare"
```

File: Cloudflare/audit_logs/tests/test_gateway_create.json [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
...8c08-d905c6e1be76\",\"name\":\"Redirect twitter\",\"description\":\"\",\"precedence\":14000,\"enabled\":true,\"action\":\"l4_override\",\"filters\":[\"l4\"],\"created_at\":\"2023-02-24T16:32:38Z\",\"updated_at\":\"2023-02-24T16:32:38Z\",\"deleted_at\":null,\"conditions\":[{\"id\":\"e588f14e-017b-4484-93e0-02a5746ededa\",\"type\":\"traffic\",\"expression\":{\"==\":{\"lhs\":\"net.dst.ip\",\"rhs\":{\"ip\":\"104.244.42.65\"}}},\"created_at\":\"2023-02-24T16:32:38Z\",\"updated_at\":\"2023-02-24T16:32:38Z\",\"deleted_at\":null}],\"traffic\":\"net.dst.ip == 104.244.42.65\",\"identity\":\"\",\"device_posture\":\"\",\"version\":1,\"rule_settings\":{\"block_page_enabled\":false,\"block_reason\":\"\",\"override_ips\":null,\"override_host\":\"\",\"l4override\":{\"ip\":\"145.239.192.199\"},\"biso_admin_controls\":{\"dp\":false,\"dcp\":false,\"dd\":false,\"du\":false,\"dk\":false,\"dcr\":false},\"add_headers\":{},\"ip_categories\":false,\"check_session\":null,\"insecure_disable_dnssec_validation\":false}},\"OldValue\":null,\"OwnerID\":\"1d1e650b3385b95db72bba7cfb1287e9\",\"ResourceID\":\"09cec157-0de7-4143-8c08-d905c6e1be76\",\"ResourceType\":\"gateway.rule\",\"When\":\"2023-02-24T16:32:38Z\"}",
```
Comment (squioc) [inline Cloudflare/audit_logs/tests/test_gateway_create.json:12]:
```suggestion
    "message": "{\"ActionResult\":true,\"ActionType\":\"gateway_create\",\"ActorEmail\":\"john.doe@example.org\",\"ActorID\":\"042fb38530054d63921f9ca81a33d5d2\",\"ActorIP\":\"::ffff:78.197.123.35\",\"ActorType\":\"user\",\"ID\":\"f6773c1b-c332-4d33-8a5b-3e7f6fafde46\",\"Interface\":\"UI\",\"Metadata\":{\"account_id\":\"1d1e650b3385b95db72bba7cfb1287e9\",\"actor_email\":\"john.doe@example.org\"},\"NewValue\":{\"id\":\"09cec157-0de7-4143-8c08-d905c6e1be76\",\"name\":\"Redirect twitter\",\"description\":\"\",\"precedence\":14000,\"enabled\":true,\"action\":\"l4_override\",\"filters\":[\"l4\"],\"created_at\":\"2023-02-24T16:32:38Z\",\"updated_at\":\"2023-02-24T16:32:38Z\",\"deleted_at\":null,\"conditions\":[{\"id\":\"e588f14e-017b-4484-93e0-02a5746ededa\",\"type\":\"traffic\",\"expression\":{\"==\":{\"lhs\":\"net.dst.ip\",\"rhs\":{\"ip\":\"104.244.42.65\"}}},\"created_at\":\"2023-02-24T16:32:38Z\",\"updated_at\":\"2023-02-24T16:32:38Z\",\"deleted_at\":null}],\"traffic\":\"net.dst.ip == 104.244.42.65\",\"identity\":\"\",\"device_posture\":\"\",\"version\":1,\"rule_settings\":{\"block_page_enabled\":false,\"block_reason\":\"\",\"override_ips\":null,\"override_host\":\"\",\"l4override\":{\"ip\":\"145.239.192.199\"},\"biso_admin_controls\":{\"dp\":false,\"dcp\":false,\"dd\":false,\"du\":false,\"dk\":false,\"dcr\":false},\"add_headers\":{},\"ip_categories\":false,\"check_session\":null,\"insecure_disable_dnssec_validation\":false}},\"OldValue\":null,\"OwnerID\":\"1d1e650b3385b95db72bba7cfb1287e9\",\"ResourceID\":\"09cec157-0de7-4143-8c08-d905c6e1be76\",\"ResourceType\":\"gateway.rule\",\"When\":\"2023-02-24T16:32:38Z\"}"
```

File: Cloudflare/audit_logs/tests/test_gateway_create.json [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    "@timestamp": "2023-02-24T16:32:38Z",
+    "user": {
+      "email": "sebastien.quioc@sekoia.fr",
```
Comment (squioc) [inline Cloudflare/audit_logs/tests/test_gateway_create.json:15]:
```suggestion
      "email": "john.doe@example.org",
```

File: Cloudflare/audit_logs/tests/test_gateway_create.json [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+      "id": "4fbb09dc367b74f992786681022426c2"
```
Comment (squioc) [inline Cloudflare/audit_logs/tests/test_gateway_create.json:16]:
```suggestion
      "id": "042fb38530054d63921f9ca81a33d5d2"
```

File: Cloudflare/audit_logs/tests/test_gateway_create_location.json [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+          "client_default": false,
+          "linked_policies": null,
+          "name": "AWS VM",
+          "networks": [
+            {
+              "network": "15.188.186.81/32"
+            }
+          ]
+        },
+        "account_id": "1d1e650b3385b95db72bba7cfb1287e9",
```
Comment (squioc) [inline Cloudflare/audit_logs/tests/test_gateway_create_location.json:34]: I think this information may be duplicated in [`cloud.account.id`](https://www.elastic.co/guide/en/ecs/8.6/ecs-cloud.html#field-cloud-account-id)

File: Cloudflare/audit_logs/tests/test_gateway_create_location.json [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+        "action_detail": "create gateway location",
+        "actor_email": "sebastien.quioc@sekoia.fr",
```
Comment (squioc) [inline Cloudflare/audit_logs/tests/test_gateway_create_location.json:36]: Could you redact the actor_email?

File: Cloudflare/audit_logs/tests/test_job_create.json [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+      "id": "1d1e650b3385b95db72bba7cfb1287e9"
+    },
+    "cloudflare": {
+      "ActionResult": true,
+      "ActionType": "job_create",
+      "ActorType": "user",
+      "ID": "abc4831a-502a-4e05-a11f-7151dbce21a8",
+      "Interface": "API",
+      "Metadata": {
+        "actor_email": "sebastien.quioc@sekoia.fr"
```
Comment (squioc) [inline Cloudflare/audit_logs/tests/test_job_create.json:24]: Could you redact the actor_email?

File: Cloudflare/audit_logs/tests/test_job_update.json [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
...inTLSHandshakeDurationMs,ParentRayID,RequestHeaders,ResponseHeaders,SecurityLevel,SmartRouteColoID,UpperTierColoID,WAFAction,WAFAttackScore,WAFFlags,WAFMatchedVar,WAFProfile,WAFRCEAttackScore,WAFRuleID,WAFRuleMessage,WAFSQLiAttackScore,WAFXSSAttackScore,WorkerCPUTime,WorkerStatus,WorkerSubrequest,WorkerSubrequestCount,WorkerWallTimeUs,ZoneName\\u0026timestamps=rfc3339\\u0026CVE-2021-44228=false\",\"destination_conf\":\"https://intake.sekoia.io/plain/batch\",\"enabled\":true,\"logstream\":true,\"frequency\":\"high\",\"kind\":\"\",\"max_upload_bytes\":5000000,\"max_upload_records\":1000},\"OldValue\":{\"id\":183719,\"dataset\":\"http_requests\",\"name\":\"\",\"logpull_options\":\"fields=ClientIP,ClientRequestHost,ClientRequestMethod,ClientRequestURI,EdgeEndTimestamp,EdgeResponseBytes,EdgeResponseStatus,EdgeStartTimestamp,RayID\\u0026timestamps=rfc3339\",\"destination_conf\":\"https://intake.sekoia.io/plain/batch\",\"enabled\":true,\"logstream\":true,\"frequency\":\"high\",\"kind\":\"\",\"max_upload_bytes\":5000000,\"max_upload_records\":1000},\"OwnerID\":\"1d1e650b3385b95db72bba7cfb1287e9\",\"ResourceID\":\"183719\",\"ResourceType\":\"logpush job\",\"When\":\"2023-02-24T17:00:39Z\"}"
```
Comment (squioc) [inline Cloudflare/audit_logs/tests/test_job_update.json:9]: Could you redact the actor_email and actor_id?

File: Cloudflare/audit_logs/tests/test_job_update.json [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
...nTLSHandshakeDurationMs,ParentRayID,RequestHeaders,ResponseHeaders,SecurityLevel,SmartRouteColoID,UpperTierColoID,WAFAction,WAFAttackScore,WAFFlags,WAFMatchedVar,WAFProfile,WAFRCEAttackScore,WAFRuleID,WAFRuleMessage,WAFSQLiAttackScore,WAFXSSAttackScore,WorkerCPUTime,WorkerStatus,WorkerSubrequest,WorkerSubrequestCount,WorkerWallTimeUs,ZoneName\\u0026timestamps=rfc3339\\u0026CVE-2021-44228=false\",\"destination_conf\":\"https://intake.sekoia.io/plain/batch\",\"enabled\":true,\"logstream\":true,\"frequency\":\"high\",\"kind\":\"\",\"max_upload_bytes\":5000000,\"max_upload_records\":1000},\"OldValue\":{\"id\":183719,\"dataset\":\"http_requests\",\"name\":\"\",\"logpull_options\":\"fields=ClientIP,ClientRequestHost,ClientRequestMethod,ClientRequestURI,EdgeEndTimestamp,EdgeResponseBytes,EdgeResponseStatus,EdgeStartTimestamp,RayID\\u0026timestamps=rfc3339\",\"destination_conf\":\"https://intake.sekoia.io/plain/batch\",\"enabled\":true,\"logstream\":true,\"frequency\":\"high\",\"kind\":\"\",\"max_upload_bytes\":5000000,\"max_upload_records\":1000},\"OwnerID\":\"1d1e650b3385b95db72bba7cfb1287e9\",\"ResourceID\":\"183719\",\"ResourceType\":\"logpush job\",\"When\":\"2023-02-24T17:00:39Z\"}",
```
Comment (squioc) [inline Cloudflare/audit_logs/tests/test_job_update.json:12]: Could you redact the actor_email and actor_id?

File: Cloudflare/audit_logs/tests/test_job_update.json [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+      "id": "1d1e650b3385b95db72bba7cfb1287e9"
+    },
+    "cloudflare": {
+      "ActionResult": true,
+      "ActionType": "job_update",
+      "ActorType": "user",
+      "ID": "90c1f5cb-f7e0-48bf-9021-14cb2102a2e1",
+      "Interface": "UI",
+      "Metadata": {
+        "actor_email": "sebastien.quioc@sekoia.fr",
```
Comment (squioc) [inline Cloudflare/audit_logs/tests/test_job_update.json:24]: Could you redact the actor_email?

**Review by squioc — CHANGES_REQUESTED** [review]

File: Cloudflare/audit_logs/ingest/parser.yml [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+          observer.type: "proxy"
+          observer.vendor: "Cloudflare"
+          source.ip: '{{json_event.message.ActorIP}}'
+          user.email: '{{json_event.message.ActorEmail}}'
+          user.id: '{{json_event.message.ActorID}}'
+
+  set_cloudflare_fields:
+    actions:
+      - set:
+          cloudflare.ActionType: '{{json_event.message.ActionType}}'
```
Comment (squioc) [inline Cloudflare/audit_logs/ingest/parser.yml:33]: I wonder if `event.action` is not suitable for this information.

**Review by squioc — APPROVED** [review]

File: Cloudflare/audit_logs/ingest/parser.yml (hunk 11 above)
Comment (Adamowoc) [reply Cloudflare/audit_logs/ingest/parser.yml]: You're right, I'll change this

**Review by squioc — APPROVED** [review]

**Pull request merged**